api_app/analyzers_manager/file_analyzers/doc_info.py

```diff
@@ -168,6 +172,25 @@ def analyze_for_follina_cve(self) -> List[str]:
                         hits += re.findall(r"mhtml:(https?://.*?)!", target)
         return hits
 
+    def analyze_for_cve(self) -> Dict:
+        pattern = r"CVE-\d{4}-\d{4,7}"
+        cve = {}
+        ole = olefile.OleFileIO(self.filepath)
+        for entry in sorted(ole.listdir(storages=True)):
+            clsid = ole.getclsid(entry)
+            if clsid_text := KNOWN_CLSIDS.get(clsid.upper(), None):
+                if matches := re.findall(pattern, clsid_text):
+                    for match in matches:
+                        if match in cve:
+                            if clsid in cve[match]:
+                                cve[match][clsid].append(clsid_text)
+                                cve[match][clsid] = list(set(cve[match][clsid]))  # uniq
+                            else:
+                                cve[match][clsid] = [clsid_text]
+                        else:
+                            cve[match] = {clsid: [clsid_text]}
+        return cve
```

Comment:
I would take all texts where there is at least the "CVE" word, then I would extract the CVE numbers if they are available. In this way you can extract even the cases when the descripion just says "probable related to CVEs..."